pkg/xen-tools/initrd/chroot2.c

```diff
@@ -20,6 +26,10 @@ static char child_stack[STACK_SIZE];    /* Space for child's stack */
 static int childFunc(void *args)
 {
     clone_args *parsed_args = (clone_args *)args;
+
+    chroot(parsed_args->chroot);
+    chdir(parsed_args->workdir);
+
     mount("proc", "/proc", "proc", 0, NULL);
 
     setgid(parsed_args->gid);
```

Comment:
not part of this PR, but better check the error for setgid/setuid,  and also set supplementary groups too (initgroups). I will open another PR for that.